src/WorkbenchExtension.ts

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+
+export class WorkbenchExtension {
+  // tslint:disable-next-line: no-any
+  private static extension: vscode.Extension<any>|undefined;
+
+  private constructor(context: vscode.ExtensionContext) {
+    // Get extensionId from package.json
+    const packageJsonPath = context.asAbsolutePath('./package.json');
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+    const extensionId = packageJson.publisher + '.' + packageJson.name;
+
+    if (!extensionId) {
+      throw new Error('Fail to get extensionId from package.json.');
+    }
+    WorkbenchExtension.extension = vscode.extensions.getExtension(extensionId);
+  }
+
+  static getExtension(context: vscode.ExtensionContext):
+      // tslint:disable-next-line: no-any
+      vscode.Extension<any>|undefined {
+    if (!WorkbenchExtension.extension) {
+      // tslint:disable-next-line: no-unused-expression
+      new WorkbenchExtension(context);
```

Comment:
Is it better to name the method as init() instead of constructor since the actual work is initialize extension variable. Or you can directly put the initialization logic into getExtension method

Reply:
Fixed.